Simplify the logic in /assemble

chasenlehara opened the pull request:
Previously, the logic inside `assemble` had two paths that did similar things: render text templates, set download headers, send the file, etc.

Now, the logic for getting the templates remains different, but rendering and sending the file is the same.

This also includes:

- An assertion that `answers` is provided (it can be an empty object, but it has to be provided)
- Changes to `combinePdfFiles` so it doesn’t create the writer when it’s only passed one PDF that won’t be modified
- Happy-path tests for rendering text templates to PDF

Part of https://github.com/CCALI/a2jdat/issues/94

mitch-mitchel commented on src/routes/assemble.js:
```diff
@@ -61,8 +61,12 @@ if (config) {
 
+  if (!answers) {
+    return res.status(400)
+      .send('You must provide answers')
```
should this say something like `at least an empty set of answers` or `You must provide an answers model/object`? This message is really for us, so it probably doesn't matter that much.

tobiasnteireho replied:
won't this work as long as there is an answers key even if it is empty. This seems correct if it will fail on no anwsers key only, but clarifying `at least an empty set` might be better.

chasenlehara replied:
I expanded the validation logic so there’s an even better error messages returned.

mitch-mitchel commented:
one nitpick comment above, but :shipit:

Commit:
Simplify the logic in /assemble

Previously, the logic inside `assemble` had two paths that did similar things: render text templates, set download headers, send the file, etc.

Now, the logic for getting the templates remains different, but rendering and sending the file is the same.

This also includes:

- An assertion that `answers` is provided (it can be an empty object, but it has to be provided)
- Changes to `combinePdfFiles` so it doesn’t create the writer when it’s only passed one PDF that won’t be modified
- Happy-path tests for rendering text templates to PDF

Part of https://github.com/CCALI/a2jdat/issues/94

mitch-mitchel reviewed (approved):
crushed
